queries.sql

```diff
@@ -129,3 +129,94 @@ LEFT JOIN animals ON owners.id = animals.owner_id
 GROUP BY owners.full_name
 ORDER BY animal_count DESC
 LIMIT 1;
+
+/* Query Questions */
+
+-- Who was the last animal seen by William Tatcher?
+
+SELECT a.name AS william_last
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+WHERE vt.name = 'William Tatcher'
+ORDER BY v.date_of_visit DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+
+SELECT COUNT(DISTINCT v.animal_id) AS mendez_no
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+
+SELECT vt.name AS vet_name, s.name AS specialties
+FROM vets vt
+LEFT JOIN specializations sp ON vt.id = sp.vet_id
+LEFT JOIN species s ON sp.species_id = s.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+SELECT a.name AS animal_name
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+WHERE vt.name = 'Stephanie Mendez'
+  AND v.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+
+-- What animal has the most visits to vets?
+
+SELECT a.name AS animal_name, COUNT(v.animal_id) AS visit_count
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+GROUP BY a.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+
+-- Who was Maisy Smith's first visit?
+
+SELECT a.name AS first_visit_animal, MIN(v.date_of_visit) AS first_visit_date
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY a.name
+ORDER BY first_visit_date
+LIMIT 1;
+
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+
+SELECT a.name AS animal_name, vt.name AS vet_name, v.date_of_visit AS last_visit_date
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+ORDER BY v.date_of_visit DESC
+LIMIT 1;
+
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT COUNT(*) AS non_specialized_visits
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+LEFT JOIN specializations s ON vt.id = s.vet_id AND a.species_id = s.species_id
+WHERE s.species_id IS NULL;
+
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+
+SELECT s.name AS required_specialty, COUNT(*) AS visit_count
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+JOIN specializations sp ON vt.id = sp.vet_id
+JOIN species s ON sp.species_id = s.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY s.name
+ORDER BY visit_count DESC
+LIMIT 1;
```

Comment:
You did well with the Queries, kindly note that all your "JOINS" in this PR should be a "LEFT JOIN" since we're selecting the results left table and some other matching records in the other table 😄

Reply:
😄